ethereum/contracts/BasicInboundChannel.sol

```diff
@@ -4,21 +4,40 @@ pragma experimental ABIEncoderV2;
 
 import "@openzeppelin/contracts/math/SafeMath.sol";
 import "./InboundChannel.sol";
+import "./LightClientBridge.sol";
 
 contract BasicInboundChannel is InboundChannel {
     uint256 constant public MAX_GAS_PER_MESSAGE = 100000;
+    LightClientBridge public lightClientBridge;
 
-    constructor() {
+    constructor(LightClientBridge _lightClientBridge) {
         nonce = 0;
+        lightClientBridge = _lightClientBridge;
     }
 
     // TODO: Submit should take in all inputs required for verification,
     // including eg: _parachainBlockNumber, _parachainMerkleProof, parachainHeadsMMRProof
-    function submit(Message[] calldata _messages, bytes32 _commitment)
+    function submit(
+        Message[] calldata _messages,
+        bytes32 _commitment,
+        bytes32 _parachainMerkleLeaf,
+        uint256 _parachainMerkleLeafIndex,
+        uint256 _parachainMerkleLeafCount,
+        bytes32[] memory _parachainMerkleProof
+    )
         public
         override
     {
         verifyMessages(_messages, _commitment);
+        require(
```

Comment:
Just a clarification/FYI: 
```
(1) beefy payload (which is an MMRRoot) --contains-->
  (2) Leaves corresponding to new relay chain blocks  --contains-->
    (3) Single, specific relay chain leaf at index for specific relay chain block --contains-->
      (4) hash of all parachain headers in that relay chain block --contains-->
        (5) parachain header for our specific parachain at specific block --contains-->
          (6)  _commitment in our parachain at that block ->
            (7) _messages in that commitment
```

 - we want to get verification all from (7) -> (1)
 - (7) -> (6) is done already, as in old code
 - This current PR and calls you've added looks like they're intended for verifying (3) -> (1), right? Overall looks good.
 - We still need to verify (6) -> (3) - can do in future PR though